api/python/t4/packages.py

```diff
@@ -721,3 +729,34 @@ def _materialize(self, path):
             # Treat as a local path
             pkg.set(logical_key, new_entry)
         return pkg
+
+    def diff(self, other_pkg):
+        """
+        Returns three lists -- added, modified, deleted.
+
+        Added: present in other_pkg but not in self.
```

Comment:
Looks like you're logic is the reverse of these case, added is in self_keys but not other and deleted is in other but not self.